src/request.rs

```diff
@@ -236,6 +245,9 @@ impl Request {
                         None
                     }
                     content_length => {
+                        if request_line.method == Method::Get {
+                            return Err(RequestError::InvalidRequest);
```

Comment:
I think we should have a separate error for this, something like `GetWithBody`. I'm open to other naming suggestions.